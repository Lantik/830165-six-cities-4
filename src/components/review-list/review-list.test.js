import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ReviewList from './review-list.jsx';
import Review from '../review/review.jsx';
import {doWithoutPropTypes} from '../../utils/testing/testing.js';

const mock = {
  reviews: [
    {id: 0}, 
    {id: 1}, 
    {id: 2}, 
    {id: 3},
    {id: 4},
    {id: 5},
    {id: 6},
    {id: 7},
    {id: 8},
    {id: 9},
    {id: 10},
    {id: 11},
    {id: 12},
    {id: 13}
  ]
};

it(`<ReviewList /> rendered correctly`, () => doWithoutPropTypes([Review], () => {
  const {reviews} = mock;
  const renderer = new ShallowRenderer();

  renderer.render(<ReviewList reviews={reviews} />);
  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
}));
