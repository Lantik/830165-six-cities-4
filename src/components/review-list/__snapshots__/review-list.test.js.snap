// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList /> rendered correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 0,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 1,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 2,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 3,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 4,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 5,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 6,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 7,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 8,
        }
      }
    />
  </li>
  <li
    className="reviews__item"
  >
    <Review
      review={
        Object {
          "id": 9,
        }
      }
    />
  </li>
</ul>
`;
